### PR TITLE
[CDAP-13111] - Fix performance issue with pipeline detailed view

### PR DESCRIPTION
**Pull request by ajainarayanan**
#### Issue:
- We had `$watch` inside the `link` function in `node-metrics` directive which was causing too many DOM manipulations inside the watch. 

From my investigation this is what was causing the lag when users opened a huge pipeline

#### Fix:
- The `$watch` was needed inside the link function to remove some nodes based on the metrics data (metric being too large remove the labels). 
- Instead of doing that we now set a flag to `true` or `false` and use `ng-if` to show and hide the labels.
- Fixes run level info to not overlap with the DAG.

screen shot 2018-02-27 at 12 25 19 pm

JIRA: https://issues.cask.co/browse/CDAP-13111

#### Note:
https://issues.cask.co/browse/CDAP-13112 is still not fixed. Because of this, the navigation between the studio and detailed view (clone) is not smooth as we wanted but this fix requires some help from the backend and we will defer that fix for another PR.

### CODE REVIEW

File: cdap-ui/app/directives/node-metrics/node-metrics.less [hunk 1]
```diff
@@ -22,7 +22,6 @@
   font-size: 11px;
   overflow: hidden;
   white-space: nowrap;
-  text-overflow: ellipsis;
```
Comment (tonybach) [inline cdap-ui/app/directives/node-metrics/node-metrics.less]: I think this shouldn't be removed. We should still show ellipses if the metrics still overflow even after we have removed the labels

**Comment by ajainarayanan** [comment]
@tonybach I couldn't entirely get why we were adding commas to the number and ellipsis to the beginning of the number.
Right now this is how we show metrics,

screen shot 2018-02-28 at 7 35 11 pm

screen shot 2018-02-28 at 7 34 37 pm

If this is what we require I might as well simplify the `commaSeparatedNumber` to just a parseInt and convert to locale string. Let me know if I am missing something.

**Comment by tonybach** [comment]
Talked with @ajainarayanan, he will discuss with @leacd and @bdmogal about the positioning of the ellipses.

**Comment by ajainarayanan** [comment]
Talked to @leacd @bdmogal and we decided for now to add ellipsis at the end like normal text. We will revisit the problem of addressing change in the numbers for each node later in 5.0

**Review by elfenheart — APPROVED** [review]
👍 code LGTM

**Pull request merged**